kratos/utilities/assign_mpcs_to_neighbours_utility.h

```diff
@@ -0,0 +1,252 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:         BSD License
+//                   Kratos default license: kratos/license.txt
+//
+//  Main author:     Sebastian Ares de Parga Regalado
+//
+
+#pragma once
+
+// System includes
+#include <string>
+#include <iostream>
+
+// External includes
+
+// Include kratos definitions
+#include "includes/define.h"
+
+// Application includes
+#include "utilities/variable_utils.h"
+#include "utilities/rbf_shape_functions_utility.h"
+
+// Project includes
+#include "spatial_containers/spatial_search.h"
+#include "spatial_containers/bins_dynamic.h"
+#include "utilities/builtin_timer.h"
+
+namespace Kratos {
+///@addtogroup Kratos Core
+///@{
+
+///@name Kratos Classes
+///@{
+
+/**
+ * @class AssignMPCsToNeighboursUtility
+ * @ingroup Kratos Core
+ * @brief Assing MPCs to Neighbouring Nodes
+ * @details This class provides a method to search for neighbouring nodes of one node
+ * within a given radius and assign a multipoint constraint (MPC) using radial basis functions.
+ * @author Sebastian Ares de Parga Regalado
+ */
+
+class KRATOS_API(KRATOS_CORE) AssignMPCsToNeighboursUtility
+{
+    public:
+      ///@name Type Definitions
+      ///@{
+      
+      //Node Types
+      using NodeType = ModelPart::NodeType;
+
+      //Dof types
+      using DofType = NodeType::DofType;
+
+      //Configure Types
+      using NodesContainerType = ModelPart::NodesContainerType;
+
+      //Bin
+      using NodeBinsType = BinsDynamic<3, NodeType, NodesContainerType::ContainerType>;
+
+      /// General containers type definitions
+      using ConstraintContainerType = ModelPart::MasterSlaveConstraintContainerType;
+      using ResultNodesContainerType = NodesContainerType::ContainerType;
+      using VectorResultNodesContainerType = std::vector<ResultNodesContainerType>;
+      using RadiusArrayType = std::vector<double>;
+
+      using DofPointerVectorType = std::vector< Dof<double>::Pointer >;
+
+      /// Pointer definition of AssignMPCsToNeighboursUtility
+      KRATOS_CLASS_POINTER_DEFINITION(AssignMPCsToNeighboursUtility);
+
+      ///@}
+      ///@name Life Cycle
+      ///@{
+
+      /// Default constructor.
```

Comment:
This is not the default constructor. Indeed, I'd delete the default constructor (it's being automatically added by the compiler).